### PR TITLE
Bumped library versions to compile with Scala 3.1.1

### PR DESCRIPTION
**Pull request by hmf**
I came across the pull request [ WIP: Support for scala 3.1 #896 ](https://github.com/almond-sh/almond/pull/896) and took a stab at it. Apparently the TASTy versions between the `2.13.*` and `3.*.*` branches needs to be paired correctly. Version `2.13.6` is compatible with `3.0.*`;  we need at least `2.13.7` for the `3.1.*` versions. Cannot recall were I read this. 

I have bumped several libraries and the code compiles for all 2.12, 2.13, 3.0 and 3.1 versions. Note that Ammonite has packages compiled for the 2.13.8 branch, however there do not seem to be any cross compilation libraries for 2.13.8. Tests also seem to run correctly.

### CODE REVIEW

**Comment by Quafadas** [comment]
See also https://github.com/almond-sh/almond/pull/902... lots of people are hoping for the same :-)!

File: project/deps.sc [hunk 1]
```diff
@@ -36,42 +36,45 @@ object Deps {
...
-  def mdoc                     = ivy"org.scalameta::mdoc:2.2.21"
+  def mdoc                     = ivy"org.scalameta::mdoc:2.3.1"
   def metabrowseServer         = ivy"org.scalameta::metabrowse-server:0.2.3"
   def scalafmtDynamic          = ivy"org.scalameta::scalafmt-dynamic:${Versions.scalafmt}"
   def scalaReflect(sv: String) = ivy"org.scala-lang:scala-reflect:$sv"
   def scalaRx                  = ivy"com.lihaoyi::scalarx:0.4.3"
-  def scalatags                = ivy"com.lihaoyi::scalatags:0.9.4"
-  def slf4jNop                 = ivy"org.slf4j:slf4j-nop:1.8.0-beta4"
+  def scalatags                = ivy"com.lihaoyi::scalatags:0.11.1"
+  def slf4jNop                 = ivy"org.slf4j:slf4j-nop:2.0.0-alpha6"
```
Comment (sideeffffect) [inline project/deps.sc:65]: This is upgrading slf4j to `2.x`. Shouldn't we stick to `1.x` instead?

Comment (hmf) [reply project/deps.sc:65]: Had issue with messages regarding not finding the slf4jNop. Also note that slf4j had a security vulnerability in 2.x but don't think it is this version. Seems to run ok. Is this whats blocking its merge?

Comment (hmf) [reply project/deps.sc:65]: Ok. Will have to update and push again.

**Comment by hmf** [comment]
I have made the slf4j changes and made a new commit (https://github.com/almond-sh/almond/pull/918/commits/687c73c3cd8f027995c35125816562778b571af2). I have reviewed the conflicts and changed the `.gitignore` accordingly. Other changes seem ok. All versions compiled on my machine. BTW, we now have Scala 3.1.2. Maybe we can update to that minor?

**Comment by alexarchambault** [comment]
Merging the remaining change, thanks @hmf!

**Pull request merged**